e2e/cephfs.go

```diff
@@ -2491,6 +2491,83 @@ var _ = Describe(cephfsType, func() {
 				}
 			})
 
+			By("verify rados objects are within a namespace", func() {
+				updateRadosNamespace := func(radosNamespaceName string) {
+					framework.Logf("updating configmap with rados namespace %s", radosNamespace)
+					radosNamespace = radosNamespaceName
+					err := deleteConfigMap(cephFSDirPath)
+					if err != nil {
+						framework.Failf("failed to delete configmap:: %v", err)
+					}
+					err = createConfigMap(cephFSDirPath, f.ClientSet, f)
+					if err != nil {
+						framework.Failf("failed to create configmap: %v", err)
+					}
+
+					// delete csi pods
+					err = deletePodWithLabel("app in (ceph-csi-cephfs, csi-cephfsplugin, csi-cephfsplugin-provisioner)",
+						cephCSINamespace, false)
+					if err != nil {
+						framework.Failf("failed to delete pods with labels: %v", err)
+					}
+					// wait for csi pods to come up
+					err = waitForDaemonSets(cephFSDeamonSetName, cephCSINamespace, f.ClientSet, deployTimeout)
+					if err != nil {
+						framework.Failf("timeout waiting for daemonset pods: %v", err)
+					}
+					err = waitForDeploymentComplete(f.ClientSet, cephFSDeploymentName, cephCSINamespace, deployTimeout)
+					if err != nil {
+						framework.Failf("timeout waiting for deployment pods: %v", err)
+					}
+				}
+
+				// radosNamespace is a global variable, so we need to save the old value
+				// and restore it after the test.
+				oldRadosNamespace := radosNamespace
+				newRadosNamespace := "cephfs-ns"
+
+				updateRadosNamespace(newRadosNamespace)
+				defer func() {
+					updateRadosNamespace(oldRadosNamespace)
+				}()
+
+				err := deleteResource(cephFSExamplePath + "storageclass.yaml")
+				if err != nil {
+					framework.Failf("failed to delete CephFS storageclass: %v", err)
+				}
+				err = createCephfsStorageClass(f.ClientSet, f, true, nil)
+				if err != nil {
+					framework.Failf("failed to create CephFS storageclass: %v", err)
+				}
+				// create a PVC and bind it to an app
+				pvc, pod, err := createPVCAndAppBinding(pvcPath, appPath, f, deployTimeout)
+				if err != nil {
+					framework.Failf("failed to validate CephFS pvc and application binding: %v", err)
+				}
+
+				validateOmapCount(f, 1, cephfsType, metadataPool, volumesType)
+
+				// delete resources
+				err = deletePod(pod.Name, pod.Namespace, f.ClientSet, deployTimeout)
```

Comment:
please add a test for snapshot and Restore pvc as well as we will use this new radosnamespace for metadata as well.